src/useWhenValueChanges.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import React from 'react'
+
+export default function useWhenValueChanges(
+  value: any,
+  callback: () => void,
+  isEqual: (any, any) => boolean = (a, b) => a === b
+) {
+  const previous = React.useRef(value)
+  React.useEffect(() => {
+    if (!isEqual(value, previous.current)) {
+      callback()
+    }
+    // We really only care if value changes, as callback doesn't change.
```

Comment:
seems to me that you have forgotten about:
```js
previous.current = value
```

Reply:
🤦‍♂️ Hooray for code review!! Will fix.